core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -130,6 +132,20 @@ class SparkHadoopUtil extends Logging {
     UserGroupInformation.loginUserFromKeytab(principalName, keytabFilename)
   }
 
+  def updateCurrentUserHDFSDelegationToken(): Unit = {
+    val conf = new Configuration()
+    val nsId = DFSUtil.getNamenodeNameServiceId(conf)
+    val isHaEnabled = HAUtil.isHAEnabled(conf, nsId)
+
+    if(isHaEnabled){
+      val ugi = UserGroupInformation.getCurrentUser
+      val uri = FileSystem.getDefaultUri(conf)
+      val map = DFSUtil.getHaNnRpcAddresses(conf)
```

Comment:
I'd try and avoid this class if possible. It is tagged as `@Private` in hadoop, and in hadoop 2.8 it'll remain in the larger hadoop-hdfs JAR, not the leaner client-side hadoop-hdfs-client JAR. The method isn't going away, and there will be a {{DFSUtilClient}} where it will move to -but that doesn't help current code.

If it can't be avoided, then the HDFS team will need to be kept up to date with its use and so encouraged not to break things